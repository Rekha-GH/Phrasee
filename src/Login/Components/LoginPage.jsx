import React, { useCallback } from 'react';
import { Form, Input, Button, Checkbox } from 'antd';
import { auth } from '../Model/LoginModel';
import { useDispatch } from 'react-redux';
import sessionSlice from '../Store/LoginBundle';
import patientsSlice from '../../Patient/Store/PatientBundle';
import { useHistory } from 'react-router-dom';

const LoginPage = React.memo(() => {
    const dispatch = useDispatch();
    const history = useHistory();
    const onFinish = useCallback((values) => {
        auth(values.username, values.password).then((data) => {
            dispatch(sessionSlice.actions.setSession(data.token));
            dispatch(patientsSlice.actions.setPatients(data.patients));
            history.push("/");
        });
    }, [dispatch, history]);

    return (
        <main>
            <Form
                layout={'vertical'}
                name="basic"
                initialValues={{ remember: true }}
                onFinish={onFinish}
            >
                <Form.Item
                    label="Username"

                    rules={[
                        {
                            required: true,
                            message: 'Please input your username!',
                        },
                        {
                            pattern: /^(\S)+@\S{1,128}\.\S{1,6}$/,
                            message: "Invalid Username",
                        },
                    ]} name="username"
                >
                    <Input
                        size="large"
                    />
                </Form.Item>

                <Form.Item
                    label="Password"
                    name="password"
                    rules={[
                        {
                            required: true,
                            message: 'Please input your password!',
                        },
                        {
                            pattern: /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,128}$/,
                            message: "minimum 8 characters, at least one letter and one number",
                        },
                    ]}
                >
                    <Input.Password
                        size="large"
                    />
                </Form.Item>

                <Form.Item name="remember" valuePropName="checked">
                    <Checkbox>Remember me</Checkbox>
                </Form.Item>

                <Form.Item>
                    <Button
                        type="primary"
                        htmlType="submit"
                        size="large"
                    >
                        Log in
                    </Button>
                </Form.Item>
            </Form>
        </main>
    );
});

LoginPage.displayName = 'LoginPage';
export default LoginPage;