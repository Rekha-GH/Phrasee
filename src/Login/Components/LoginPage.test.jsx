import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import LoginPage from './LoginPage';
import { shallow } from '../../setupTests';

describe('LoginPage Tests', () => {
  const useSelectorMock = jest.spyOn(ReactRedux, 'useDispatch').mockImplementation(() => undefined);

  test('LoginPage renders correctly', () => {
    const wrapper = shallow(<LoginPage></LoginPage>);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
