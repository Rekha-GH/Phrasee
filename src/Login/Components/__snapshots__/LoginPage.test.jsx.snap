// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Tests LoginPage renders correctly 1`] = `
<main>
  <ForwardRef(InternalForm)
    initialValues={
      Object {
        "remember": true,
      }
    }
    layout="vertical"
    name="basic"
    onFinish={[Function]}
  >
    <FormItem
      label="Username"
      name="username"
      rules={
        Array [
          Object {
            "message": "Please input your username!",
            "required": true,
          },
          Object {
            "message": "Invalid Username",
            "pattern": /\\^\\(\\\\S\\)\\+@\\\\S\\{1,128\\}\\\\\\.\\\\S\\{1,6\\}\\$/,
          },
        ]
      }
    >
      <Input
        size="large"
        type="text"
      />
    </FormItem>
    <FormItem
      label="Password"
      name="password"
      rules={
        Array [
          Object {
            "message": "Please input your password!",
            "required": true,
          },
          Object {
            "message": "minimum 8 characters, at least one letter and one number",
            "pattern": /\\^\\(\\?=\\.\\*\\[A-Za-z\\]\\)\\(\\?=\\.\\*\\\\d\\)\\[A-Za-z\\\\d\\]\\{8,128\\}\\$/,
          },
        ]
      }
    >
      <Password
        action="click"
        iconRender={[Function]}
        size="large"
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      name="remember"
      valuePropName="checked"
    >
      <Checkbox
        indeterminate={false}
      >
        Remember me
      </Checkbox>
    </FormItem>
    <FormItem>
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        size="large"
        type="primary"
      >
        Log in
      </Button>
    </FormItem>
  </ForwardRef(InternalForm)>
</main>
`;
